file_formats/cmt_format.ml

```diff
@@ -197,12 +213,169 @@ let clear_env binary_annots =
 
   else binary_annots
 
+let iter_on_usages ~index =
+  let f ~namespace env path lid =
+    let not_ghost { Location.loc = { loc_ghost; _ }; _ } = not loc_ghost in
+    if not_ghost lid then
+      try
+        let shape = Env.shape_of_path ~namespace env path in
+        let shape = Local_reduce.weak_reduce env shape in
+        if not (Shape.is_closed shape) then
+          index := (Unresolved shape, lid) :: !index
+        else Option.iter
+          (fun uid -> index := (Resolved uid, lid) :: !index)
+          shape.Shape.uid
+      with Not_found -> ()
+  in
+  let add_constructor_description env lid =
+    function
+    | { Types.cstr_tag = Cstr_extension (path, _); _ } ->
+        f ~namespace:Extension_constructor env path lid
+    | { Types.cstr_uid = Predef "()"; _ } ->
+        (* We do not index occurrences of `()` *)
+        ()
+    | { Types.cstr_uid; _ } ->
+        index := (Resolved cstr_uid, lid) :: !index
```

Comment:
Forgive the naïve question but are we sure this is resolved? If the constructor comes from a different unit, we'll get the uid from the declaration, not from the definition, right?
Does this not matter?